### DISCUSSION TITLE
Is there an automatic navigation feature for single matching directory?

### DISCUSSION THREAD

**Discussion by lvlassis** [sxyazi/yazi#1221] [1. Q&A] [answered]
### Description

Hello Yazi developers,

I am a new user of Yazi, transitioning from Ranger. One feature in Ranger that I find extremely useful is its search functionality. In Ranger, when you press 'f' to enter search mode and start typing, if the number of possible directories narrows down to just one, it automatically navigates to that directory. This feature greatly enhances navigation speed and efficiency.

### Question

Does Yazi have a similar automatic navigation feature when the search results reduce to a single directory? If not, is there any existing workaround or setting that can enable this behavior?

Thank you for your assistance and for developing such a performant file manager. I'm looking forward to making the most out of Yazi.

Best regards,
Sevenings

**Comment by sxyazi** [comment, marked as answer]
Hi, I made a plugin for it a few days ago that extends the existing filter - it will automatically enter the only directory, and I've also added other enhancements, like supporting continuous filtering and automatically opening the hovered file on submitting:


https://github.com/sxyazi/yazi/assets/17523360/0007e5e5-ed9e-4104-ad13-0678b73af1f1



Feel free to use it: https://github.com/yazi-rs/plugins/tree/main/smart-filter.yazi

You can find more awesome plugins at https://yazi-rs.github.io/docs/resources

**Reply by lvlassis** [reply]
Hi, thanks for the answer, I downloaded the plugin and I'm really enjoying the feature, but there is still one little detail that I was trying to tweak through the init.lua and am not managing to do it. 

I've noticed that the filter bar stays up after the plugin enters a directory, different from ranger, that closes the smart-filter function after it. At one side it's good for the speed, so I don't need to type `F` again and again, and it allows me to navigate through my files as it was a 'light-speed mode'. But, as I'm used to ranger, and  it closes the input when I enter a new directory, on Yazi I often end up typing a little more letters and falling in an unwanted directory. 

In that way, I started looking through the script and the yazi lua documentation, as they are very simple, if I could find a way to close the filter input box. It ended up that I've passed some time searching, messing around with some code and haven't find anything. I was searching for something similar to the `ya.manager_emit()` function, but for the input, like `ya.input_emit('close')`, that I'm suspecting that it probably doesn't even exists, and there isn't an way to control the input via plugins. 

Would you give me an help? Maybe there is an actual way to do it that I don't know, or it just need to be implemented.

**Reply by sxyazi** [reply]
> like ya.input_emit('close'), that I'm suspecting that it probably doesn't even exists

Yeah it hasn't been there yet, I'll add it later

**Reply by sxyazi** [reply]
Added in https://github.com/sxyazi/yazi/pull/1241/files#diff-279c160e96171a108456a5763499c511ca15b627fa204c270afeda8c58eadab0, now you should be able to close the input with:

```lua
ya.input_emit("close", {})
```

**Reply by lvlassis** [reply]
Thank you very much! I've just updated the smart-filter plugin with the new command and it works great, now I've achieved the behavior that I was looking for.

 I'm really grateful for the attention, and hope that the project evolves even more! You are doing an excellent job on it! Thank you!

**Reply by lpnh** [reply]
Do you mind sharing your solution? I think I will apply it to mine as well.

**Reply by lvlassis** [reply]
I don't mind, there it is my init.lua, [init.zip](https://github.com/user-attachments/files/16072191/init.zip)
I'm not used to github's chat system, it does not allow uploading .lua files for default, so that was the way that I've found to share it.

**Reply by lpnh** [reply]
No problem, thanks! I'll just leave the diff here for anyone else who would like to achieve the same thing.
(note: it requires Yazi nightly.)

```diff
 		if h.unique and h.is_dir then
 			ya.manager_emit("escape", { filter = true })
 			ya.manager_emit("enter", { h.url })
-			input = prompt()
+			ya.input_emit("close", {})
 		elseif event == 1 then
 			ya.manager_emit("escape", { filter = true })
 			ya.manager_emit(h.is_dir and "enter" or "open", { h.url })
 ```
